Mail/Views/New Message/Scheduled panel/ScheduleOptionButtonRow.swift

```diff
@@ -0,0 +1,67 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakCoreSwiftUI
+import MailCoreUI
+import SwiftUI
+
+struct ScheduleOptionButtonRow: View {
+    @Environment(\.dismiss) private var dismiss
+
+    let option: ScheduleSendOption
+    let setScheduleAction: (Date) -> Void
+
+    var body: some View {
+        if let scheduleDate = option.date {
+            Button {
+                setScheduleAction(scheduleDate)
+                dismiss()
+            } label: {
+                HStack {
+                    Label {
+                        Text(option.title)
+                            .textStyle(.body)
+                            .padding(.leading, value: .medium)
+                    } icon: {
+                        option.icon
+                            .iconSize(.large)
+                    }
+                    .frame(maxWidth: .infinity, alignment: .leading)
+
+                    Text(scheduleDate.formatted(.dateTime.weekday(.abbreviated).day().month(.abbreviated).hour().minute()))
```

Comment:
Can we declare the formatter in a variable? It is not really clean to see it in the body